src/item.cpp

```diff
@@ -1740,7 +1740,7 @@ std::string item::info(std::vector<iteminfo> &info, const iteminfo_query *parts,
             const auto &styles = g->u.ma_styles;
             const std::string valid_styles = enumerate_as_string(styles.begin(), styles.end(),
                 [this](const matype_id &mid) {
-                return mid.obj().has_weapon(typeId()) ? _( mid.obj().name ) : std::string();
+                return mid.obj().has_weapon(typeId()) ? _( mid.obj().name.c_str() ) : std::string();
```

Comment:
We now could use simple `name()` for this, or am I mistaken?

Reply:
Won't work - there is no `name()` method - only `name` field which is a `std::string` and we need `const char*`.